secret_store/src/key_server_cluster/net/accept_connection.rs

```diff
@@ -19,20 +19,21 @@ use std::sync::Arc;
 use std::net::SocketAddr;
 use std::time::Duration;
 use futures::{Future, Poll};
-use tokio_core::reactor::Handle;
-use tokio_core::net::TcpStream;
+use tokio::net::TcpStream;
 use key_server_cluster::{Error, NodeKeyPair};
 use key_server_cluster::io::{accept_handshake, Handshake, Deadline, deadline};
 use key_server_cluster::net::Connection;
 
 /// Create future for accepting incoming connection.
-pub fn accept_connection(address: SocketAddr, stream: TcpStream, handle: &Handle, self_key_pair: Arc<NodeKeyPair>) -> Deadline<AcceptConnection> {
+pub fn accept_connection(stream: TcpStream, self_key_pair: Arc<NodeKeyPair>) -> Deadline<AcceptConnection> {
+	let address = stream.peer_addr().expect("Unable to determine tcp peer address");
```

Comment:
Could you please add TODO to remove this expect in the future? Not sure that it can't fail here

Reply:
It could fail which probably means it would be better either to accept the address as a separate argument or return a result. I'll add a note to that effect.